### ISSUE TITLE
Systemd misses reload command

### ISSUE THREAD

**Issue by davromaniak** [ytti/oxidized#871]
Hi.

Quite related to issues #864 and #760, it would be usefull to add an ExecReload command to the Systemd service file.

Maybe we could copy the example in the [extra/oxidized.init.d](https://github.com/ytti/oxidized/blob/master/extra/oxidized.init.d) file `curl -s http://localhost:8888/reload?format=json -O /dev/null` or finding a way to get the "rest" variable in the config file (I would some dirty grep | awk :D) to call the good URL.

But in this case, it would need oxidized to be directly dependant from oxidized-web, so that's not the best idea from my point of view.

Another way would be a little ruby snippet to call the `load` function directly from the [lib/oxidized/nodes.rb](https://github.com/ytti/oxidized/blob/master/lib/oxidized/nodes.rb), which is what the REST API does during the /reload.

But well, I just found out that sending the HUP signal to the oxidized PID will do the trick... so something like that should work `ExecReload=/bin/kill -HUP $MAINPID`.

Thanks.

**Comment by davromaniak** [comment]
Here's what I get when I send the `kill -HUP` on the oxidized PID : 
```
- Gracefully stopping, waiting for requests to finish
=== puma shutdown: 2017-05-29 08:25:40 +0200 ===
- Goodbye!
```
It seems puma catches the signal and then kills itself :(.

Thanks.

**Comment by ytti** [comment]
Just stabbing in the dark here.

It's probably race condition now if Oxidized main thread or Puma thread gets the HUP.

We probably want the puma thread to stop accepting HUP trap:
https://ruby-doc.org/core-2.2.0/Thread.html#method-c-handle_interrupt

We start Puma here:
https://github.com/ytti/oxidized-web/blob/master/lib/oxidized/web.rb#L28

**Comment by wk** [comment]
This appears to have been resolved in #1144, can this be closed?
